tests/test_integration.py

```diff
@@ -100,22 +100,21 @@
 
 
 @pytest.mark.asyncio
-@pytest.mark.xfail  # FIXME: docker websocket seems not working as expected
 async def test_stdio_stdin(docker, testing_images, shell_container):
     ws = await shell_container.websocket(stdin=True, stdout=True, stream=True)
-    await ws.send_str('echo "hello world"\n')
-    with aiohttp.Timeout(2):
-        # TODO: fix timeout
-        resp = await ws.receive()
-    print(resp)
-    assert resp.data == b"hello world\n"
+    await ws.send_str('echo hello world\n')
+
+    resp = await ws.receive()
+    assert resp.data == "echo hello world\r\n"
     await ws.close()
-    # TODO: ensure container stopped on its own because stdin was closed
 
     # Cross-check container logs.
+    await asyncio.sleep(1)
```

Comment:
it is a good idea!  I have fix that testcase like this,  But the worst site  when expect_line not found, the stream will block until connection timeout error, and TEST FAILED!